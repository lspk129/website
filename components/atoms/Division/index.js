import styled from 'styled-components';

const Division = styled.div`
  ${props => props.textAlign && `text-align: ${props.textAlign}`};
  // margin properties
  ${props => props.m && `margin: ${props.m}rem;`}
  ${props => props.mx && `margin: 0rem ${props.mx}rem;`}
  ${props => props.my && `margin: ${props.my}rem 0rem;`}
  ${props => props.mt && `margin-top: ${props.mt}rem;`}
  ${props => props.mb && `margin-bottom: ${props.mb}rem;`}
  ${props => props.mr && `margin-right: ${props.mr}rem;`}
  ${props => props.ml && `margin-left: ${props.ml}rem;`}
  // padding properties
  ${props => props.p && `padding: ${props.p}rem;`}
  ${props => props.px && `padding: 0rem ${props.px}rem;`}
  ${props => props.py && `padding: ${props.py}rem 0rem;`}
  ${props => props.pt && `padding-top: ${props.pt}rem;`}
  ${props => props.pb && `padding-bottom: ${props.pb}rem;`}
  ${props => props.pr && `padding-right: ${props.pr}rem;`}
  ${props => props.pl && `padding-left: ${props.pl}rem;`}
  // background properties
  ${props => props.bg && `
    background-image: url(${props.src});
    background-repeat: no-repeat;
    background-attachment: fixed;
    background-size: cover;
    display: grid;
  `}
`;

export default Division;
