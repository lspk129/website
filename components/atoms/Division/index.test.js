import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import Division from './';

describe('Division', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Division />);
    expect(wrapper).toBePresent();
    expect(wrapper).toHaveTagName('div');
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('render with display props', () => {
    const wrapper = mount(<Division display={'grid'} />);
    expect(wrapper).toHaveProp('display', 'grid');
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('render with textAlign props', () => {
    const wrapper = mount(<Division textAlign={'center'} />);
    expect(wrapper).toHaveProp('textAlign', 'center');
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('render with alignSelf props', () => {
    const wrapper = mount(<Division alignSelf={'flex-end'} />);
    expect(wrapper).toHaveProp('alignSelf', 'flex-end');
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('render with margin props', () => {
    const wrapper = mount(<Division mx={'1'} />);
    expect(wrapper).toHaveProp('mx', '1');
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('render with margin left and right props', () => {
    const wrapper = mount(<Division mx={2} />);
    expect(wrapper).toHaveProp('mx', 2);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('render with margin top and bottom props', () => {
    const wrapper = mount(<Division my={3} />);
    expect(wrapper).toHaveProp('my', 3);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('render with margin top props', () => {
    const wrapper = mount(<Division mt={4} />);
    expect(wrapper).toHaveProp('mt', 4);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders with margin bottom props', () => {
    const wrapper = mount(<Division mb={5} />);
    expect(wrapper).toHaveProp('mb', 5);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders with margin right props', () => {
    const wrapper = mount(<Division mr={6} />);
    expect(wrapper).toHaveProp('mr', 6);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders with margin left props', () => {
    const wrapper = mount(<Division ml={7} />);
    expect(wrapper).toHaveProp('ml', 7);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders with padding props', () => {
    const wrapper = mount(<Division p={8} />);
    expect(wrapper).toHaveProp('p', 8);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders Division with padding left and right props', () => {
    const wrapper = mount(<Division px={7} />);
    expect(wrapper).toHaveProp('px', 7);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders with padding top and bottom props', () => {
    const wrapper = mount(<Division py={6} />);
    expect(wrapper).toHaveProp('py', 6);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders with padding top props', () => {
    const wrapper = mount(<Division pt={5} />);
    expect(wrapper).toHaveProp('pt', 5);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders with padding bottom props', () => {
    const wrapper = mount(<Division pb={4} />);
    expect(wrapper).toHaveProp('pb', 4);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders with padding right props', () => {
    const wrapper = mount(<Division pr={3} />);
    expect(wrapper).toHaveProp('pr', 3);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders with padding left props', () => {
    const wrapper = mount(<Division pl={2} />);
    expect(wrapper).toHaveProp('pl', 2);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders with background color props', () => {
    const wrapper = mount(<Division backgroundColor={'#BADA55'} />);
    expect(wrapper).toHaveProp('backgroundColor', '#BADA55');
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders with background image source props', () => {
    const wrapper = mount(<Division src={'../../../static/Service_image.jpg'} />);
    expect(wrapper).toHaveProp('src', '../../../static/Service_image.jpg');
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders with z-index props', () => {
    const wrapper = mount(<Division zIndex={10} />);
    expect(wrapper).toHaveProp('zIndex', 10);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
