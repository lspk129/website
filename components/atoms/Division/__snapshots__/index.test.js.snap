// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Division render with alignItems props 1`] = `
<styled.div
  alignItems="flex-end"
>
  <div
    className="sc-bdVaJa gtqFVS"
  />
</styled.div>
`;

exports[`Division render with alignSelf props 1`] = `
<styled.div
  alignSelf="flex-end"
>
  <div
    className="sc-bdVaJa ljFKUl"
  />
</styled.div>
`;

exports[`Division render with display props 1`] = `
<styled.div
  display="grid"
>
  <div
    className="sc-bdVaJa hZNMen"
    display="grid"
  />
</styled.div>
`;

exports[`Division render with margin left and right props 1`] = `
<styled.div
  mx={2}
>
  <div
    className="sc-bdVaJa gFdapX"
  />
</styled.div>
`;

exports[`Division render with margin props 1`] = `
<styled.div
  mx="1"
>
  <div
    className="sc-bdVaJa cPcdQu"
  />
</styled.div>
`;

exports[`Division render with margin top and bottom props 1`] = `
<styled.div
  my={3}
>
  <div
    className="sc-bdVaJa kcIRCd"
  />
</styled.div>
`;

exports[`Division render with margin top props 1`] = `
<styled.div
  mt={4}
>
  <div
    className="sc-bdVaJa QgfOx"
  />
</styled.div>
`;

exports[`Division render with textAlign props 1`] = `
<styled.div
  textAlign="center"
>
  <div
    className="sc-bdVaJa kDuLde"
  />
</styled.div>
`;

exports[`Division renders Division with padding left and right props 1`] = `
<styled.div
  px={7}
>
  <div
    className="sc-bdVaJa ilpvEk"
  />
</styled.div>
`;

exports[`Division renders correctly 1`] = `
<div
  className="sc-bdVaJa krulIk"
/>
`;

exports[`Division renders with background color props 1`] = `
<styled.div
  backgroundColor="#BADA55"
>
  <div
    className="sc-bdVaJa jkpNnC"
  />
</styled.div>
`;

exports[`Division renders with background image source props 1`] = `
<styled.div
  src="../../../static/Service_image.jpg"
>
  <div
    className="sc-bdVaJa krulIk"
    src="../../../static/Service_image.jpg"
  />
</styled.div>
`;

exports[`Division renders with margin bottom props 1`] = `
<styled.div
  mb={5}
>
  <div
    className="sc-bdVaJa iuEEUi"
  />
</styled.div>
`;

exports[`Division renders with margin left props 1`] = `
<styled.div
  ml={7}
>
  <div
    className="sc-bdVaJa eLUbNz"
  />
</styled.div>
`;

exports[`Division renders with margin right props 1`] = `
<styled.div
  mr={6}
>
  <div
    className="sc-bdVaJa jwQvFS"
  />
</styled.div>
`;

exports[`Division renders with padding bottom props 1`] = `
<styled.div
  pb={4}
>
  <div
    className="sc-bdVaJa iOxZvf"
  />
</styled.div>
`;

exports[`Division renders with padding left props 1`] = `
<styled.div
  pl={2}
>
  <div
    className="sc-bdVaJa gkAnra"
  />
</styled.div>
`;

exports[`Division renders with padding props 1`] = `
<styled.div
  p={8}
>
  <div
    className="sc-bdVaJa ccbJFG"
  />
</styled.div>
`;

exports[`Division renders with padding right props 1`] = `
<styled.div
  pr={3}
>
  <div
    className="sc-bdVaJa bvotjn"
  />
</styled.div>
`;

exports[`Division renders with padding top and bottom props 1`] = `
<styled.div
  py={6}
>
  <div
    className="sc-bdVaJa bfxFee"
  />
</styled.div>
`;

exports[`Division renders with padding top props 1`] = `
<styled.div
  pt={5}
>
  <div
    className="sc-bdVaJa ilHIHY"
  />
</styled.div>
`;

exports[`Division renders with z-index props 1`] = `
<styled.div
  zIndex={10}
>
  <div
    className="sc-bdVaJa eqZxnH"
  />
</styled.div>
`;
